server/sonar-server/src/main/java/org/sonar/server/notification/NotificationDaemon.java

```diff
@@ -62,9 +62,9 @@
   private ScheduledExecutorService executorService;
   private boolean stopping = false;
 
-  public NotificationDaemon(Settings settings, DefaultNotificationManager manager, NotificationService service) {
-    this.delayInSeconds = settings.getLong(PROPERTY_DELAY);
-    this.delayBeforeReportingStatusInSeconds = settings.getLong(PROPERTY_DELAY_BEFORE_REPORTING_STATUS);
+  public NotificationDaemon(Configuration config, DefaultNotificationManager manager, NotificationService service) {
+    this.delayInSeconds = config.getLong(PROPERTY_DELAY).get();
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Call "Optional#isPresent()" before accessing the value. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS3655)